internal/mode/static/telemetry/collector.go

```diff
@@ -147,3 +155,16 @@ func collectGraphResourceCount(
 
 	return ngfResourceCounts, nil
 }
+
+func (c DataCollectorImpl) collectClusterID(ctx context.Context) (string, error) {
+	kubeNS := types.NamespacedName{
+		Name: dataplane.KubeSystem,
+	}
+	var nsMetadata v1.Namespace
+	err := c.cfg.K8sClientReader.Get(ctx, kubeNS, &nsMetadata)
+	if err != nil {
+		return "", fmt.Errorf("failed to get metadata for cluster :%w", err)
+	}
+
+	return string(nsMetadata.ObjectMeta.GetUID()), nil
```

Comment:
I think we should just be able to call `nsMetadata.GetUID()`

Reply:
`nsMetadata.GetUID()` returns field of type `"k8s.io/apimachinery/pkg/types".UID`. I don't mind keep it that type but wanted to cross check with you

Reply:
It's the same function call, so they both return the same value. ObjectMeta is an embedded field on the Namespace object, so its methods propagate up to the top level resource.